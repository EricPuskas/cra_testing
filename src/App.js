import React, { useState } from "react";
import Header from "./components/Header/Header";
import Headline from "./components/Headline/Headline";
import Button from "./components/Button/Button";
import ListItem from "./components/ListItem/ListItem";
import { connect } from "react-redux";
import { getPosts } from "./actions";
import "./App.scss";

const tempArr = [
  {
    firstName: "Joe",
    lastName: "Bloggs",
    email: "jbloggs@gmail.com",
    age: 34,
    onlineStatus: true
  }
];

const App = ({ getPosts, posts }) => {
  const fetchPosts = () => {
    getPosts();
    handleUpdate();
  };
  const [hideBtn, setHideBtn] = useState(false);
  const handleUpdate = () => {
    setHideBtn(!hideBtn);
  };
  return (
    <div className="App" data-test="App">
      <Header />
      <section className="main">
        <Headline
          header="Posts"
          description="Click button to render posts."
          tempArr={tempArr}
        />
        {!hideBtn && (
          <Button
            data-test="btn"
            buttonText="Get Posts"
            emitEvent={fetchPosts}
          />
        )}
        {posts.length > 0 && (
          <div>
            {posts.map(post => (
              <ListItem key={post.id} title={post.title} desc={post.body} />
            ))}
          </div>
        )}
      </section>
    </div>
  );
};
const mapStateToProps = state => ({
  posts: state.posts
});

export default connect(
  mapStateToProps,
  { getPosts }
)(App);
