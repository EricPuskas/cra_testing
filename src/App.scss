body {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  background: #707174;
  font-family: "Ubuntu", sans-serif;
  font-size: 18px;
  line-height: 22px;
}

.main {
  width: 100%;
  margin: 50px auto 80px;
  padding: 0 10px;
}
