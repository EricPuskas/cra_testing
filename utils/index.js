import checkPropTypes from "check-prop-types";
import { applyMiddleware, createStore } from "redux";
import rootReducer from "../src/reducers";
import { middleware } from "../src/store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const checkProps = (component, expectedProps) =>
  checkPropTypes(component.propTypes, expectedProps, "props", component.name);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
