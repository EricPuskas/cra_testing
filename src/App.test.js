import App from "./App";
import { shallow } from "enzyme";
import { findByAttribute, testStore } from "../utils";
import React from "react";

const setup = (initState = {}) => {
  const store = testStore(initState);
  const wrapper = shallow(<App store={store} />)
    .childAt(0)
    .dive();
  return wrapper;
};
describe("App Component", () => {
  let wrapper;
  beforeEach(() => {
    const initState = {
      posts: [
        { id: 1, title: "Title 1", body: "Some Text" },
        { id: 2, title: "Title 2", body: "Some Text" },
        { id: 3, title: "Title 3", body: "Some Text" }
      ]
    };
    wrapper = setup(initState);
  });

  it("Should render without errors", () => {
    const component = findByAttribute(wrapper, "App");
    expect(component.length).toBe(1);
  });

  it("Should pass snapshot test", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
