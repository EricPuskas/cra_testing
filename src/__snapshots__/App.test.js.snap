// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Should pass snapshot test 1`] = `"<div class=\\"App\\" data-test=\\"App\\"><header data-test=\\"header\\"><div class=\\"wrap\\"><div class=\\"logo\\"><img src=\\"logo.png\\" alt=\\"logo\\" data-test=\\"logo\\"/></div><div class=\\"logo-title\\">CRA-Testing</div></div></header><section class=\\"main\\"><div data-test=\\"Headline\\"><h1 data-test=\\"header\\"> Posts </h1><p data-test=\\"description\\"> Click button to render posts. </p></div><button data-test=\\"SharedBtn\\">Get Posts</button><div><div data-test=\\"listItem\\"><h2 data-test=\\"title\\">Title 1</h2><p data-test=\\"desc\\"> Some Text </p></div><div data-test=\\"listItem\\"><h2 data-test=\\"title\\">Title 2</h2><p data-test=\\"desc\\"> Some Text </p></div><div data-test=\\"listItem\\"><h2 data-test=\\"title\\">Title 3</h2><p data-test=\\"desc\\"> Some Text </p></div></div></section></div>"`;
